logic_tree_tools.py

```diff
@@ -0,0 +1,972 @@
+
+    modified_logic_tree_pair.source_logic_tree = copy.deepcopy(
+        modified_source_logic_tree
+    )
+    modified_logic_tree_pair.ground_motion_logic_tree = copy.deepcopy(
+        modified_ground_motion_logic_tree
+    )
+
+    modified_logic_tree_pairs.append(modified_logic_tree_pair)
+
+    return modified_logic_tree_pairs
+
+
+def print_info_about_logic_tree(logic_tree: Union[SourceLogicTree, GMCMLogicTree]):
+    """
+    Prints information about a logic tree.
+
+    This function prints details about the type of logic tree (SourceLogicTree or GMCMLogicTree),
+    the number of branch sets it contains, and information about each branch set.
+
+    Parameters
+    ----------
+    logic_tree : Union[SourceLogicTree, GMCMLogicTree]
+        The logic tree for which information is to be printed.
+    """
+
+    print("")  # Add a blank line for readability
+
+    if isinstance(logic_tree, SourceLogicTree):
+        print("Logic tree is a SourceLogicTree")
+
+    if isinstance(logic_tree, GMCMLogicTree):
+        print("Logic tree is a GMCMLogicTree")
+
+    print(f"Logic tree has {len(logic_tree.branch_sets)} branch sets")
+
+    for branch_set_index, branch_set in enumerate(logic_tree.branch_sets):
+
+        print(
+            f"Branch set index {branch_set_index} has name {branch_set.long_name} ({branch_set.short_name}) and contains {len(branch_set.branches)} branches"
+        )
+
+
+def print_info_about_logic_tree_pairs(
+    logic_tree_pairs: Union[list[CustomLogicTreePair], CustomLogicTreePair]
+):
+    """
+    Print information about a logic tree pair or a list of logic tree pairs.
+
+    Parameters
+    ----------
+    logic_tree_pairs : list[CustomLogicTreePair] or CustomLogicTreePair
+        The logic_tree_pairs to print information about.
+    """
+
+    if not isinstance(logic_tree_pairs, list):
+        logic_tree_pairs = [logic_tree_pairs]
+
+    print(f"Printing information about {len(logic_tree_pairs)} logic tree pairs")
+
+    for logic_tree_pair_index, logic_tree_pair in enumerate(logic_tree_pairs):
+        print()
+        print(f"Logic tree at index {logic_tree_pair_index}:")
+        print(
+            f"source_logic_tree_note: {logic_tree_pairs[logic_tree_pair_index].source_logic_tree_note}"
+        )
+        print(
+            f"ground_motion_logic_tree_note: {logic_tree_pairs[logic_tree_pair_index].ground_motion_logic_tree_note}"
+        )
+
+        if logic_tree_pair.source_logic_tree is not None:
+            print_info_about_logic_tree(logic_tree_pair.source_logic_tree)
+
+        if logic_tree_pair.ground_motion_logic_tree is not None:
+            print_info_about_logic_tree(logic_tree_pair.ground_motion_logic_tree)
+
+
+def get_logic_tree_pairs_for_tectonic_selection(
+    initial_logic_tree_pair: CustomLogicTreePair,
+    tectonic_region_type_sets: list[list[str]],
+    which_interfaces: list[str],
```

Comment:
Again, enums. Would also simplify the types probably.

Reply:
And the argument names I would again also change.

Reply:
Changed the argument name tectonic_region_type_sets to tectonic_region_type_groups to avoid potential confusion with the set type